tests/unit/unit_test_multipart_upload.py

```diff
@@ -114,3 +116,36 @@ def chunk_upload(part):
 
         # assert _put_chunk was called at least once
         assert_greater_equal(len(mocked_put_chunk.call_args_list), 1)
+
+
+def test__multipart_upload():
+    mocked_get_chunk_function = MagicMock(side_effect=[1, 2, 3, 4])
+    file_size = 1*MB
+    filepath = make_bogus_binary_file(n=file_size)
+    md5 = md5_for_file(filepath).hexdigest()
+    status = {'partsState': {},
+              'uploadId': {},
+              'state': 'COMPLETED'}
+    with patch.object(syn, "restPOST", return_value=status),\
+            patch.object(SingleThreadPool, "map", return_value=None) as single_thread_pool_map,\
```

Comment:
these patch.objects would no longer be needed. just make the pool_provider.get_pool return a mock object